### PR TITLE
add job version to solve version compatibility issues for version update

### PR DESCRIPTION
**Pull request by hongyh13**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/FhirToDataLakeAzureStorageJobInfo.cs [hunk 1]
```diff
@@ -71,5 +72,13 @@ private string TryParseAsFhirToDataLakeProcessingJobInputData()
 
             return null;
         }
+
+        private static string RemoveJobVersionFromInput(object inputData)
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/FhirToDataLakeAzureStorageJobInfo.cs:76]: Maybe we can rename to GetIdentifierText() and comment why we need to exclude version here.

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/Jobs/FhirToDataLakeAzureStorageJobInfoTests.cs [hunk 2]
```diff
@@ -159,5 +161,38 @@ public void GivenTwoSameDefinitionsForJobVersionV1AndV2_WhenGetJobIdentifier_The
...
+                Result = string.Empty,
+                CancelRequested = false,
+                CreateDate = DateTime.UtcNow,
+                HeartbeatDateTime = DateTime.UtcNow,
+            };
+
+            var jobject = JObject.FromObject(orchestratorDefinition);
+            jobject[JobVersionManager.JobVersionKey].Parent.Remove();
+
+            Assert.Equal(jobject.ToString().ComputeHash(),jobInfo.JobIdentifier());
```
Comment (moria97) [inline FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/Jobs/FhirToDataLakeAzureStorageJobInfoTests.cs:195]: We can also add Assert.NotEqual to directly compute hash

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/FhirToDataLakeAzureStorageJobInfo.cs (hunk 1 above)
Comment (hongyh13) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/Models/FhirToDataLakeAzureStorageJobInfo.cs:76]: Sure, resolved.

File: FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/Jobs/FhirToDataLakeAzureStorageJobInfoTests.cs (hunk 2 above)
Comment (hongyh13) [reply FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/Jobs/FhirToDataLakeAzureStorageJobInfoTests.cs]: Added.

**Review by moria97 — APPROVED** [review]

**Pull request merged**